src/compute/protocol.rs

```diff
@@ -159,6 +159,20 @@ pub struct ServerFlavor {
     pub vcpu_count: u32,
 }
 
+fn bool_from_config_drive_string<'de, D>(deserializer: D) -> Result<bool, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    match String::deserialize(deserializer)?.as_ref() {
+        "True" => Ok(true),
+        "" => Ok(false),
+        other => Err(de::Error::invalid_value(
+            de::Unexpected::Str(other),
+            &"True or empty",
```

Comment:
Do you really need an ampersand before the string slice? I did not try, but looks redundant at first glance.

Reply:
```
error[E0277]: the trait bound `str: serde::de::Visitor<'_>` is not satisfied
   --> src/compute/protocol.rs:171:13
    |
171 |             "True or empty",
    |             ^^^^^^^^^^^^^^^ the trait `serde::de::Visitor<'_>` is not implemented for `str`
    |
    = note: required because of the requirements on the impl of `serde::de::Expected` for `str`
    = note: required for the cast to the object type `dyn serde::de::Expected`
```

Reply:
Life is learning..